src/tests/uat/nsx_remediation_uat.py

```diff
@@ -0,0 +1,157 @@
+
+    Args:
+        cb (CBCloudAPI): The API connection object.
+        ids (list[int]: List of IDs explicitly selected. If empty, all devices will be selected in random order.
+
+    Returns:
+        list[Device]: A list of possible devices to use for the test.
+    """
+    query = cb.select(Device)
+    base_devices = [d for d in query if d.nsx_available]
+    devices = []
+    if not base_devices:
+        raise RuntimeError("ERROR: no devices in org with NSX available")
+    print(f"{len(base_devices)} device(s) available for testing")
+    if ids:
+        for v in ids:
+            devices.extend([d for d in base_devices if d.id == v])
+    else:
+        devices = random.sample(base_devices, k=len(base_devices))
+    if not devices:
+        raise RuntimeError("ERROR: no devices selected for testing")
+    print(f"{len(devices)} device(s) selected for testing")
+    return devices
+
+
+def test_one_combo(device, tag, toggle):
+    """
+    Run an NSX remediation test on a specific combination of arguments.
+
+    Args:
+        device (Device): The device under test.
+        tag (str): The NSX tag being set.
+        toggle (bool): True if the tag is being set, False if it's being removed.
+
+    Returns:
+        bool: True if the test completed (possibly with errors printed), False if the device was not suitable for test.
+    """
+    print(f"Testing ({tag}, {toggle}) on device #{device.id}")
+    try:
+        job = device.nsx_remediation(tag, toggle)
+        if job is None:
+            print(f"Setting ({tag}, {toggle}) on device #{device.id} was a no-op")
+            return True
+        job.await_result()
+        device.refresh()
+        if toggle and device.nsx_distributed_firewall_policy != tag:
+            print(f"ERROR: after ({tag}, {toggle}) set, expected {tag}, got {device.nsx_distributed_firewall_policy}")
+        elif toggle is False and device.nsx_distributed_firewall_policy is not None:
+            print(f"ERROR: after ({tag}, {toggle}) set, expected None, got {device.nsx_distributed_firewall_policy}")
+        else:
+            print(f"Setting ({tag}, {toggle}) on device #{device.id} OK")
+    except NSXJobError:
+        print(f"...Device #{device.id} not suitable for testing or erroneous state, try again")
+        return False
+    except ServerError as e:
+        print(f"ERROR: setting ({tag}, {toggle}) on device #{device.id} failed: {e}")
+    except ApiError as e:
+        print(f"ERROR: setting ({tag}, {toggle}) on device #{device.id} failed: {e}")
+    return True
+
+
+def test_device(device):
+    """
+    Run a NSX test sequence on the specified device.
+
+    Args:
+        device (Device): The device under test.
+
+    Returns:
+        bool: True if the test completed (possibly with errors printed), False if the device was not suitable for test.
+    """
+    print(f"Beginning test on device #{device.id}")
+    before_tag = device.nsx_distributed_firewall_policy
+    if before_tag:
```

Comment:
Can you add some comments indicating what is being performed? I can't easily tell what is being set versus unset?